services/tables/src/main/java/com/linkedin/openhouse/tables/toggle/FeatureToggleAspect.java

```diff
@@ -0,0 +1,95 @@
+package com.linkedin.openhouse.tables.toggle;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.linkedin.openhouse.internal.catalog.toggle.IcebergFeatureGate;
+import com.linkedin.openhouse.tables.config.TblPropsToggleRegistry;
+import com.linkedin.openhouse.tables.model.TableDto;
+import com.linkedin.openhouse.tables.repository.impl.PreservedPropsToggleEnabler;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.aspectj.lang.JoinPoint;
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.annotation.Before;
+import org.aspectj.lang.reflect.MethodSignature;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * To use {@link TableFeatureToggle} one can add implementation of feature-toggle checking in this
+ * class with different types of advises injected (e.g. {@link Before}).
+ */
+@Aspect
+@Component
+@Slf4j
+public class FeatureToggleAspect {
+
+  @Autowired private TableFeatureToggle tableFeatureToggle;
+
+  @Autowired private TblPropsToggleRegistry tblPropsToggleRegistry;
+
+  @Around("@annotation(preservedPropsToggleEnabler)")
+  public boolean checkTblPropEnabled(
+      ProceedingJoinPoint proceedingJoinPoint,
+      PreservedPropsToggleEnabler preservedPropsToggleEnabler) {
+    if (((MethodSignature) proceedingJoinPoint.getSignature()).getReturnType() == boolean.class
+        && proceedingJoinPoint.getArgs()[1] instanceof TableDto
+        && proceedingJoinPoint.getArgs()[0] instanceof String) {
+      TableDto tableDto = (TableDto) proceedingJoinPoint.getArgs()[1];
+      String key = (String) proceedingJoinPoint.getArgs()[0];
+
+      return tblPropsToggleRegistry
+          .obtainFeatureByKey(key)
+          .map(
+              feature ->
+                  evalProceeding(proceedingJoinPoint)
+                      && !tableFeatureToggle.isFeatureActivated(
```

Comment:
this condition will only work if we have the feature as `allowMOR` or `allowSomething`. It wouldn't be able to achieve `disallowSomething` for example say `disallowPoliciesForTable`. I understand that its not in the scope, but can we add an **important note in TblsProsToggleRegistry documentation saying only allowlisting is supported**?

Reply:
yes the semantics of this annotation is VERY strict. 
At this point I think annotation is really just a syntactical sugar to avoid contaminating the main body of service implementation, not a general enough approach to extend to other semantics.

Reply:
> At this point I think annotation is really just a syntactical sugar to avoid contaminating the main body 

thats alright! but lets use documentation as a key to convey strictness

Reply:
can we add such documentation to `TblsProsToggleRegistry`?

Reply:
sure, addressed altogether with the comment here. https://github.com/linkedin/openhouse/pull/189#discussion_r1772175096